src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -48,7 +47,16 @@ fun createLookupElement(element: RsCompositeElement, scopeName: String): LookupE
             .withTailText(element.valueParameterList?.text?.replace("\\s+".toRegex(), " ") ?: "()")
             .appendTailText(element.extraTailText, true)
             .withInsertHandler handler@ { context: InsertionContext, _: LookupElement ->
-                if (context.isInUseBlock) return@handler
+                if (context.isInUseBlock) {
+                    val offset = context.tailOffset - 1
+                    val useItem = PsiTreeUtil.findElementOfClassAtOffset(context.file, offset, RsUseItem::class.java, false)
+                    val hasSemicolon = useItem!!.lastChild!!.elementType != RsElementTypes.SEMICOLON
```

Comment:
Hm, now that I think of it, could we get rid of those `!!` in `useItem!!`? What if `isInUseBlock` returned not a `Bool`, but a `RsUseItem?`? So that we can compare it with `null`.

Reply:
I refactored it to your suggestion.